test/health_cards/health_card_test.rb

```diff
@@ -95,44 +95,67 @@ class HealthCardTest < ActiveSupport::TestCase
     assert_equal(resource_nums, inc_array)
   end
 
+  test 'changes to stripped bundle do not affect bundle values' do
+    stripped_bundle = @health_card.strip_fhir_bundle
+
+    # Check bundle
+    stripped_bundle.type = 'foobar'
+    assert_not_equal @health_card.bundle.type, stripped_bundle.type
+
+    # Check entry value
+    patient = stripped_bundle.entry[0].resource
+    patient.name[0].family = 'foobar'
+    assert_not_equal @health_card.bundle.entry[0].resource.name[0].family,
+                     stripped_bundle.entry[0].resource.name[0].family
+
+    # Check entries
+    stripped_bundle.entry = []
+    assert_not_equal @health_card.bundle.entry.length, stripped_bundle.entry.length
```

Comment:
I think a decent test here would be to convert the bundle to a JSON string, strip the bundle, convert the original bundle to a JSON string again and make sure they're the same.